lib/algolia/index.rb

```diff
@@ -224,36 +236,39 @@ def get_object(objectID, attributesToRetrieve = nil)
     #
     # @param objectIDs the array of unique identifier of the objects to retrieve
     # @param attributesToRetrieve (optional) if set, contains the list of attributes to retrieve as an array of strings of a string separated by ","
+    # @param request_options contains extra parameters to send with your query
     #
-    def get_objects(objectIDs, attributesToRetrieve = nil)
+    def get_objects(objectIDs, attributesToRetrieve = nil, request_options = {})
       attributesToRetrieve = attributesToRetrieve.join(',') if attributesToRetrieve.is_a?(Array)
       requests = objectIDs.map do |objectID|
         req = {:indexName => name, :objectID => objectID.to_s}
         req[:attributesToRetrieve] = attributesToRetrieve unless attributesToRetrieve.nil?
         req
       end
-      client.post(Protocol.objects_uri, { :requests => requests }.to_json, :read)['results']
+      client.post(Protocol.objects_uri, { :requests => requests }.to_json, :read, request_options)['results']
     end
 
     # Check the status of a task on the server.
     # All server task are asynchronous and you can check the status of a task with this method.
     #
     # @param taskID the id of the task returned by server
+    # @param request_options contains extra parameters to send with your query
     #
-    def get_task_status(taskID)
-      client.get(Protocol.task_uri(name, taskID), :read)["status"]
+    def get_task_status(taskID, request_options = {})
+      client.get(Protocol.task_uri(name, taskID), :read, request_options)['status']
     end
 
     # Wait the publication of a task on the server.
     # All server task are asynchronous and you can check with this method that the task is published.
     #
     # @param taskID the id of the task returned by server
     # @param timeBeforeRetry the time in milliseconds before retry (default = 100ms)
+    # @param request_options contains extra parameters to send with your query
     #
-    def wait_task(taskID, timeBeforeRetry = 100)
+    def wait_task(taskID, timeBeforeRetry = 100, request_options = {})
```

Comment:
Here I would leave it like this. We also have `def add_object(obj, objectID = nil, request_options = {})` for instance.

If I do this, I should as well do this for each and every method.